src/distilabel/llm/anthropic.py

```diff
@@ -0,0 +1,124 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from pydantic import PrivateAttr, SecretStr
+
+from distilabel.llm.base import AsyncLLM
+
+if TYPE_CHECKING:
+    from anthropic import AsyncAnthropic
+
+    from distilabel.llm.typing import GenerateOutput
+    from distilabel.steps.task.typing import ChatType
+
+
+class AnthropicLLM(AsyncLLM):
+    """Anthropic LLM implementation running the Async API client.
+
+    Args:
+        model: the model name to use for the LLM.
+        api_key: the API key to authenticate the requests to the Anthropic API.
+        base_url: the base URL to use for the Anthropic API. Defaults to `None` which means that `https://api.anthropic.com` will be used internally.
+        http_client: the HTTP client to use for the Anthropic API. Defaults to None.
+        timeout: the maximum time in seconds to wait for a response. Defaults to 600.0.
+        max_retries: the maximum number of retries for the LLM. Defaults to 2.
+    """
+
+    model: str = "claude-3-opus-20240229"
+    api_key: Optional[SecretStr] = os.getenv("ANTHROPIC_API_KEY", None)  # type: ignore
+    base_url: Union[str, None] = None
+    timeout: float = 600.0
+    http_client: Union[str, None] = None
+    max_retries: int = 2
+
+    _env_var: Optional[str] = PrivateAttr(default="ANTHROPIC_API_KEY")
+    _aclient: Optional["AsyncAnthropic"] = PrivateAttr(...)
+
+    def load(self, api_key: Optional[str] = None) -> None:
+        """Loads the `AsyncAnthropic` client to use the Anthropic async API."""
+        try:
+            from anthropic import AsyncAnthropic
+        except ImportError as ie:
+            raise ImportError(
+                "Anthropic Python client is not installed. Please install it using"
+                " `pip install anthropic`."
+            ) from ie
+
+        self.api_key = self._handle_api_key_value(
+            self_value=self.api_key,
+            load_value=api_key,
+            env_var=self._env_var,  # type: ignore
+        )
+
+        self._aclient = AsyncAnthropic(
+            api_key=self.api_key.get_secret_value(),
+            base_url=self.base_url,
+            timeout=self.timeout,
+            http_client=self.http_client,
+            max_retries=self.max_retries,
+        )
+
+    @property
+    def model_name(self) -> str:
+        """Returns the model name used for the LLM."""
+        return self.model
+
+    async def agenerate(  # type: ignore
+        self,
+        input: "ChatType",
+        system: str = "",
```

Comment:
We can remove this parameter, as it would be part of `input`:

```python
[
    {"role": "system", "content": "..."},
    {"role": "user", "content": "..."}
]
```